src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,39 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  static addAnimal(animal) {
+    this.alive.push(animal);
+  }
+
+  constructor(name, health = 100) {
+    this.health = health;
+    this.name = name;
+
+    Animal.addAnimal(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  hidden = false;
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    const isHerbivore = animal instanceof Herbivore;
+    const isHidden = animal.hidden;
+
+    if (isHerbivore && !isHidden) {
+      animal.health -= 50;
+
+      Animal.alive = Animal.alive.filter((aliveAnimal) => !!aliveAnimal.health);
```

Comment:
Using filter inside the bite method is not efficient because it will loop through all alive animals every time a bite occurs. Consider moving this logic outside of the bite method and only invoke it when necessary, for example, after all bites in a sequence have been performed.